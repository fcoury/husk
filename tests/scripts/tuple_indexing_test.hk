// Test tuple index accessors
let t1 = (1, 2, 3);
let t2 = (4.5, "hello", true);
let nested = ((1, 2), (3, 4));

// Basic access
println!("t1.0 = {}", t1.0);
println!("t1.1 = {}", t1.1);
println!("t1.2 = {}", t1.2);

println!("t2.0 = {}", t2.0);
println!("t2.1 = {}", t2.1);
println!("t2.2 = {}", t2.2);

// Nested access
println!("nested.0.0 = {}", nested.0.0);
println!("nested.0.1 = {}", nested.0.1);
println!("nested.1.0 = {}", nested.1.0);
println!("nested.1.1 = {}", nested.1.1);

// In expressions
let sum = t1.0 + t1.1 + t1.2;
println!("Sum of t1 elements: {}", sum);

// Test completed - mutation tests removed as 'let mut' is not supported yet