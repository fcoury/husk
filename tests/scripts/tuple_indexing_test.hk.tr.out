t1.0 = 1
t1.1 = 2
t1.2 = 3
t2.0 = 4.5
t2.1 = hello
t2.2 = true
nested.0.0 = 1
nested.0.1 = 2
nested.1.0 = 3
nested.1.1 = 4
Sum of t1 elements: 6
